classes/controller/like.php

```diff
@@ -0,0 +1,182 @@
+        if(isloggedin()) {
+
+            /**
+            * Checking if user has already liked the post
+            */
+            
+            $sql = "SELECT COUNT(id) AS likescounter FROM " . self::ACTIONS_TABLE . " WHERE postid = ? AND userid = ?";
+            $sqlreturn = $DB->get_record_sql($sql,
+                [
+                    'postid' => $postid,
+                    'userid' => $USER->id,
+                ]
+            );
+
+            if(!empty($sqlreturn->likescounter)) {
+                $result->content = get_string('toomanylikes','local_hsuforum_actions');
+                return $result;
+            }
+
+            // Insert new action record
+            $record = (object)[];
+            $record->postid = $postid;
+            $record->userid = $USER->id;
+            $record->action = self::ACTION;
+            $record->created = time();
+            $actionid = $DB->insert_record('hsuforum_actions', $record, true);
+
+            //Get post to return
+            $sql = "
+            SELECT
+                p.id,
+                p.discussion
+            FROM
+                " . self::TABLE . " p
+            WHERE
+                p.id = $postid
+            ";
+
+            $post = $DB->get_records_sql($sql);
+
+            hsuforum_populate_post_actions($post);
+
+            $result->result = true;
+            $result->content = $post;
+
+        } else {
+            $result->result = false;
+            $result->content = 'Your session has timed out. Please login again.';
+        }
+
+        return $result;
+
+	}
+
+    /**
+     * function delete_action
+     * Remove action 'like' from database
+     * unliking a post.
+     * @param $postid
+     * @return object
+     **/ 
+	public function delete_action($postid) {
+        global $DB, $CFG, $USER;
+
+        require_once($CFG->dirroot.'/mod/hsuforum/hsuforum_actions_lib.php');
+
+        $result = (object)[];
+        $result->result = false; // set in case uncaught error happens
+        $result->content = 'Unknown error';
+
+        //Only allow to remove action if logged in
+        if(isloggedin()) {
+
+            $deleted = $DB->delete_records('hsuforum_actions', array('postid' => $postid, 'userid' => $USER->id, 'action' => self::ACTION));
+            
+            //Get post to return
+            $sql = "
+            SELECT
+                p.id,
+                p.discussion
+            FROM
+                " . self::TABLE . " p
+            WHERE
+                p.id = $postid
+            ";
+
+            $post = $DB->get_records_sql($sql);
+
+            hsuforum_populate_post_actions($post);
+
+            $result->result = true;
+            $result->content = $post;
+
+        } else {
+            $result->result = false;
+            $result->content = 'Your session has timed out. Please login again.';
+        }
+
+        return $result;
+	}
+
+    /**
+     * 
+     **/
+    public function render_action() {}
```

Comment:
Empty function - believe this will be added later

Reply:
yes